Plugins/GeoModel/src/GeoModelDetectorSurfaceFactory.cpp

```diff
@@ -6,16 +6,20 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+// Needs to be included first to avoid some conflict between forward declare and
+// typedef from Geomodel side
+// clang-format off
+#include "Acts/Plugins/GeoModel/GeoModelTree.hpp"
```

Comment:
This seems funky. Can you elaborate what this is for?

Reply:
So actually, we need to include on the top the `ReadGeomode.hpp `(which is included in the GeomodelTree.hpp) because it has the `#define _GeoShapePersistification_On_` that if it is not defined before including the GeoKernel files then it goes and does the forward declararion of the `class Persistifer ` and creates the conflict declaration